modules/gapi/test/gapi_gpu_test.cpp

```diff
@@ -202,6 +210,243 @@ TEST(GPU, Symm7x7_test)
         EXPECT_EQ(out_mat_gapi.size(), sz);
     }
 }
-#endif
+
+#ifdef HAVE_DIRECTX
+#ifdef HAVE_D3D11
+
+void createD3D11Device(ID3D11Device** device) {
+    IDXGIFactory* adapter_factory = nullptr;
+    HRESULT hr = CreateDXGIFactory(__uuidof(IDXGIFactory),
+                                    reinterpret_cast<void**>(&adapter_factory));
+    if (FAILED(hr)) {
+        CV_Error(cv::Error::BadCallBack, "Cannot create CreateDXGIFactory, error: " +
+                    std::to_string(HRESULT_CODE(hr)));
+    }
+    IDXGIAdapter* intel_adapter = nullptr;
+    UINT adapter_index = 0;
+    const unsigned int refIntelVendorID = 0x8086; // Intel
+    IDXGIAdapter* out_adapter = nullptr;
+
+    while (adapter_factory->EnumAdapters(adapter_index, &out_adapter) != DXGI_ERROR_NOT_FOUND) {
+        DXGI_ADAPTER_DESC adapter_desc{};
+        out_adapter->GetDesc(&adapter_desc);
+        if (adapter_desc.VendorId == refIntelVendorID) {
+            intel_adapter = out_adapter;
+            break;
+        }
+        out_adapter->Release();
+        ++adapter_index;
+    }
+    if (!intel_adapter) {
+        throw SkipTestException("No Intel GPU adapter on aboard.");
+    }
+    intel_adapter->Release();
```

Comment:
why do you release it? you should use this adapter  as device below

```
 HRESULT err = D3D11CreateDevice(intel_adapter, D3D_DRIVER_TYPE_UNKNOWN,   <-----
```

Reply:
Fixed. Stupid mistake.